spec.html

```diff
@@ -40586,27 +40444,12 @@ <h1>Properties of the Proxy Constructor</h1>
         <li>has the following properties:</li>
       </ul>
 
-      <emu-clause id="sec-proxy.revocable">
+      <emu-clause id="sec-proxy.revocable" oldids="sec-proxy-revocation-functions">
         <h1>Proxy.revocable ( _target_, _handler_ )</h1>
         <p>The `Proxy.revocable` function is used to create a revocable Proxy object. When `Proxy.revocable` is called with arguments _target_ and _handler_, the following steps are taken:</p>
         <emu-alg>
           1. Let _p_ be ? ProxyCreate(_target_, _handler_).
-          1. Let _steps_ be the algorithm steps defined in <emu-xref href="#sec-proxy-revocation-functions" title></emu-xref>.
-          1. Let _length_ be the number of non-optional parameters of the function definition in <emu-xref href="#sec-proxy-revocation-functions" title></emu-xref>.
-          1. Let _revoker_ be ! CreateBuiltinFunction(_steps_, _length_, *""*, &laquo; [[RevocableProxy]] &raquo;).
-          1. Set _revoker_.[[RevocableProxy]] to _p_.
-          1. Let _result_ be ! OrdinaryObjectCreate(%Object.prototype%).
-          1. Perform ! CreateDataPropertyOrThrow(_result_, *"proxy"*, _p_).
-          1. Perform ! CreateDataPropertyOrThrow(_result_, *"revoke"*, _revoker_).
-          1. Return _result_.
-        </emu-alg>
-
-        <emu-clause id="sec-proxy-revocation-functions">
-          <h1>Proxy Revocation Functions</h1>
-          <p>A Proxy revocation function is an anonymous built-in function that has the ability to invalidate a specific Proxy object.</p>
-          <p>Each Proxy revocation function has a [[RevocableProxy]] internal slot.</p>
-          <p>When a Proxy revocation function is called, the following steps are taken:</p>
-          <emu-alg>
+          1. Let _revokerClosure_ be a new Abstract Closure with no parameters that captures nothing and performs the following steps when called:
```

Comment:
Why can't this capture _p_? The closure doesn't need to change _p_, only mutates its slots.

Reply:
It does: it sets _p_ to `null` once the proxy has been revoked, so it can bail early if called twice, in `d. Set F.[[RevocableProxy]] to null.`

I suppose since this operation is idempotent we could just do away with the early bailing, but I was trying to stick with the existing logic as closely as possible.

Reply:
Oh wow, this terrible GH review UI! When I expanded the unchanged lines to compare, it skipped the line that sets it to null! Notice the missing lines in the attached screenshot.
<img width="1171" alt="Screen Shot 2021-06-23 at 6 54 16 PM" src="https://user-images.githubusercontent.com/95697/123190478-71694d80-d454-11eb-9d09-aca118734aa3.png">

Reply:
Anyways, then everything lgtm.

Reply:
Yup, for large documents that button silently fails in a way almost indistinguishable from success. And it has done for literally years with no sign they intend to fix it are even aware it's an issue. /shrug